integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java

```diff
@@ -29,73 +31,83 @@ public BridgeSteps(TestContext context) {
         assertThat(response.getHref()).isNotNull();
         assertThat(response.getSubmittedAt()).isNotNull();
 
-        context.setRandomBridgeName(bridgeName);
-        context.setBridgeId(response.getId());
+        BridgeContext bridgeContext = context.newBridge(testBridgeName, systemBridgeName);
+        bridgeContext.setBridgeId(response.getId());
+        bridgeContext.setBridgeName(response.getName());
     }
 
-    @Then("get list of Bridge instances with access token contains Bridge with randomly generated name")
-    public void testRandomBridgeExists() {
+    @And("^the list of Bridge instances contains the Bridge \"([^\"]*)\"+$")
+    public void listOfBridgeInstancesContainsBridge(String testBridgeName) {
+        BridgeContext bridgeContext = context.getBridge(testBridgeName);
+
         BridgeListResponse response = BridgeResource.getBridgeList(context.getManagerToken());
-        assertThat(response.getItems()).anyMatch(b -> b.getName().equals(context.getRandomBridgeName()));
+
+        assertThat(response.getItems()).anyMatch(b -> b.getId().equals(bridgeContext.getBridgeId()));
+        BridgeResponse bridge = response.getItems().stream().filter(b -> b.getId().equals(bridgeContext.getBridgeId()))
+                .findFirst().orElseThrow();
+        assertThat(bridge.getName()).isEqualTo(bridgeContext.getBridgeName());
     }
 
-    @Then("^get Bridge with access token exists in status \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
-    public void bridgeDeployedWithinMinutes(String status, int timeoutMinutes) {
+    @And("^the Bridge \"([^\"]*)\" exists with status \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
+    public void bridgeExistsWithStatusWithinMinutes(String testBridgeName, String status, int timeoutMinutes) {
+        BridgeContext bridgeContext = context.getBridge(testBridgeName);
+
         Awaitility.await()
                 .atMost(Duration.ofMinutes(timeoutMinutes))
                 .pollInterval(Duration.ofSeconds(5))
                 .untilAsserted(
                         () -> BridgeResource
-                                .getBridgeDetailsResponse(context.getManagerToken(), context.getBridgeId())
+                                .getBridgeDetailsResponse(context.getManagerToken(), bridgeContext.getBridgeId())
                                 .then()
                                 .body("status", Matchers.equalTo(status))
-                                .body("endpoint", Matchers.containsString(context.getBridgeId())));
+                                .body("endpoint", Matchers.containsString(bridgeContext.getBridgeId())));
 
-        BridgeUtils.getOrRetrieveBridgeEndpoint(context);
+        BridgeUtils.getOrRetrieveBridgeEndpoint(context, testBridgeName);
     }
 
-    @When("delete a Bridge")
-    public void testDeleteBridge() {
-        BridgeResource.deleteBridge(context.getManagerToken(), context.getBridgeId());
+    @When("^the Bridge \"([^\"]*)\" is deleted$")
```

Comment:
Same as above related to active sentence

Reply:
updated